sdks/java/harness/src/main/java/org/apache/beam/fn/harness/control/ProcessBundleHandler.java

```diff
@@ -318,23 +328,15 @@ private void createRunnerAndConsumersForPTransformRecursively(
         // Add all checkpointed residuals to the response.
         response.addAllResidualRoots(bundleProcessor.getSplitListener().getResidualRoots());
 
-        // TODO(BEAM-6597): This should be reporting monitoring infos using the short id system.
-        // Get start bundle Execution Time Metrics.
-        response.addAllMonitoringInfos(
-            bundleProcessor.getStartFunctionRegistry().getExecutionTimeMonitoringInfos());
-        // Get process bundle Execution Time Metrics.
-        response.addAllMonitoringInfos(
-            bundleProcessor.getpCollectionConsumerRegistry().getExecutionTimeMonitoringInfos());
-        // Get finish bundle Execution Time Metrics.
-        response.addAllMonitoringInfos(
-            bundleProcessor.getFinishFunctionRegistry().getExecutionTimeMonitoringInfos());
-        // Extract MonitoringInfos that come from the metrics container registry.
-        response.addAllMonitoringInfos(
-            bundleProcessor.getMetricsContainerRegistry().getMonitoringInfos());
-        // Add any additional monitoring infos that the "runners" report explicitly.
-        for (ProgressRequestCallback progressRequestCallback :
-            bundleProcessor.getProgressRequestCallbacks()) {
-          response.addAllMonitoringInfos(progressRequestCallback.getMonitoringInfos());
+        // Add all metrics to the response.
+        Map<String, ByteString> monitoringData = monitoringData(bundleProcessor);
+        if (runnerAcceptsShortIds) {
+          response.putAllMonitoringData(monitoringData);
+        } else {
+          for (Map.Entry<String, ByteString> metric : response.getMonitoringDataMap().entrySet()) {
```

Comment:
for (Map.Entry<String, ByteString> metric : monitoringData.entrySet()) ?

Reply:
Oh, yes. (Leftover from refactoring.) Done.